pkg/user/parallel_tx_submission.go

```diff
@@ -0,0 +1,497 @@
+package user
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"cosmossdk.io/x/feegrant"
+	"github.com/celestiaorg/celestia-app/v6/pkg/appconsts"
+	"github.com/celestiaorg/go-square/v3/share"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	bank "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// SubmissionJob represents a transaction submission task for parallel processing
+type SubmissionJob struct {
+	Blobs    []*share.Blob
+	Options  []TxOption
+	Ctx      context.Context
+	ResultsC chan SubmissionResult
+}
+
+// SubmissionResult contains the result of a parallel transaction submission
+type SubmissionResult struct {
+	Signer     string
+	TxResponse *TxResponse
+	Error      error
+}
+
+// txQueue manages parallel transaction submission
+type txQueue struct {
```

Comment:
per request, all unnecessary stuff was unexported. also renamed this since we use it for most tx submission now.